### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: examples/EXAMPLES_LICENSE [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Zero-Clause BSD License
+
+Copyright (C) 2021 Toitware ApS
```
Comment (kasperl) [inline examples/EXAMPLES_LICENSE]: 2022

File: examples/example.toit [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the EXAMPLES_LICENSE file.
+
+import net
+import net.udp as net_udp
+import net.modules.udp
```
Comment (kasperl) [inline examples/example.toit]: Drop this import and use net.open.udp_open instead.

File: examples/example.toit [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import dns_simple_server show SimpleDns
+
+main:
+  socket := udp.Socket "0.0.0.0" 5353
```
Comment (kasperl) [inline examples/example.toit]: Use net.open.

File: src/dns_simple_server.toit [hunk 4]
```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+import binary show BIG_ENDIAN
+import bytes show Buffer
+import net.modules.udp
```
Comment (kasperl) [inline src/dns_simple_server.toit]: Drop this import.

File: src/dns_simple_server.toit [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        a_length := BIG_ENDIAN.uint16 query position + 8
+        position += 10 + a_length
+        // Currently we don't do anything with the optional data.
+        // We might want to recognize type 41, which is OPT, and allows
+        // the max UDP size of the sender to be recorded.  RFC 2671.
+      return response.get
+    return null
+
+class ResponseBuilder_:
+  substring_cache := {:}
```
Comment (kasperl) [inline src/dns_simple_server.toit]: Should this really be public? ... and does it ever change?

File: src/dns_simple_server.toit [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  packet /Buffer := Buffer
```
Comment (kasperl) [inline src/dns_simple_server.toit:108]: Should this really be public? ... and does it ever change?

File: src/dns_simple_server.toit [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+/**
+A very simple DNS server.
+The server can respond to A record queries for IPv4.
+There is a static map of domain names to IP addresses, and an
+  optional default value that is returned for any other queries.
+It should be enough for a captive portal.
+*/
+class SimpleDns:
+  hosts_ := {:}
```
Comment (kasperl) [inline src/dns_simple_server.toit]: ::=

File: src/dns_simple_server.toit [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+          packet.write prefix
+          name = name[dot + 1..]
+        else:
+          packet.write_byte name.size
+          packet.write name
+          name = ""
+
+  resource_record -> none
+      name/string
+      --address /net.IpAddress? = null
```
Comment (kasperl) [inline src/dns_simple_server.toit:153]: Sometime you have space before / in here, but not always. It looks inconsistent.

File: src/dns_simple_server.toit [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+import net.modules.dns
+
+/**
+A very simple DNS server.
+The server can respond to A record queries for IPv4.
+There is a static map of domain names to IP addresses, and an
+  optional default value that is returned for any other queries.
+It should be enough for a captive portal.
+*/
+class SimpleDns:
```
Comment (kasperl) [inline src/dns_simple_server.toit]: DnsSimpleServer or SimpleDnsServer?

File: src/dns_simple_server.toit (hunk 4 above)
Comment (erikcorry) [reply src/dns_simple_server.toit]: Currently I need this for the constants and helper functions - we could consider moving them.

**Pull request merged**